src/upstream_patches_ui.sh

```diff
@@ -204,7 +271,9 @@ function show_series_details()
       for option in "${selected_options[@]}"; do
         case "$option" in
           'Bookmark')
-            printf 'TODO' # TODO
+            create_loading_screen_notification "Bookmarking patch(es)"$'\n'"- ${patch_title}"
+            download_series "$total_patches" "$patch_url" "${lore_config['download_to']}" "$patch_title"
+            add_series_to_bookmark "${target_patch}" "${lore_config['download_to']}"
```

Comment:
If any of these functions fails, you should check and provide a message to the user that lets them know that the operation failed for some reason.

Reply:
+1 again in this comment.